app/src/main/res/layout/location_row.xml

```diff
@@ -35,7 +36,7 @@
         android:layout_marginStart="32dp"
         android:layout_marginTop="8dp"
         android:layout_marginEnd="32dp"
-        android:text="TextView"
+        android:text="@string/textview"
```

Comment:
2 suggestions here:
1) Each TextView here has the text "TextView"  That's not adding much value.  Shouldn't they describe the input they expect?
2) Better to use the hint property than text.  Text is default text in a widget; a user has to delete it.  A hint is more like a label that is embedded in the TextView, and it goes away when that TextView gets focus.  So, you don't need a separate label component.